### ISSUE TITLE
Add serialize methods to objects

### ISSUE THREAD

**Issue by justincy** [rootsdev/familysearch-javascript-sdk#88]
The biggest reason is that the objects now contain circular references so plain `JSON.stringify()` will throw an error. You can get around that using the `replacer` argument of `stringify` like this:

```js
JSON.stringify(object, function(key, value){
  if(key.indexOf('$') === 0){
    return;
  }
  return value;
});
```

But it would be much better if the SDK did that for us.

**Comment by justincy** [comment]
Mixing with [`toJSON`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior) provides an ideal solution.

```js
FS.BaseClass.prototype.toJSON = function(){
  return JSON.stringify(this, function(key, value){
    if(key.indexOf('$') === 0){
      return;
    }
    return value;
  });
};
```

**Comment by justincy** [comment]
I think we could avoid the error by removing the `client` property from the `Plumbing` object since that property is never used. But I still think we should add the `toJSON` function to the base class so that `JSON.stringify()` doesn't add junk properties when trying to serialize the SDK instance and helpers.

**Comment by justincy** [comment]
:disappointed: It's not that easy; the code I shared two weeks ago doesn't work.

1. We don't do protoype inheritance properly. We use the BaseClass constructor but don't inherit it's prototype. We would either need to refactor all classes to properly inherit or add the `toJSON` method in the constructor.

2. The return value of `toJSON` is an object, not a string.

3. Calling `JSON.stringify` on `this` inside of `toJSON` is recursive and blows the stack.

**Comment by justincy** [comment]
I think the best option is to use that code and just rename it to `serialize`:

```js
FS.BaseClass.prototype.serialize= function(){
  return JSON.stringify(this, function(key, value){
    if(key.indexOf('$') === 0){
      return;
    }
    return value;
  });
};
```

That fixes problems (2) and (3). If we don't fix protoype inheritance then we can just assign it in the constructor (not ideal but leaving the option open in case fixing inheritance is too much work).

**Comment by justincy** [comment]
closed by fc1336ff95f2021ce767912934d43597ab46f158
